src/client/ducks/repositories/selectors/index.js

```diff
@@ -72,3 +73,27 @@ export const getRepositoryId = createSelector(
     _.findKey(entities, (v) => (v.fullName === repoName))
 )
 
+export const getChangesetList = (state: Object) => _.values(state.entities.changesets)
+
+export const getChangesetsWithAvatars = (state: Object) => createSelector(
+  getChangesetList, userEntitiesSelector,
+  (changesets, users) =>
+    _.values(changesets.map(changeset => {
+      const user = _.values(users).find(x => x.username === changeset.author)
```

Comment:
Yeah, looks too heavy, and I afraid it could bring some difficulties in the future. That is because this value coming from the mercurial, and we are not doing anything with is before exposing in API. Let me take a look what we can do with it.